plugin.php

```diff
@@ -63,11 +65,8 @@ function () {
  * Instantiate the plugin.
  */
 function main(): void {
-	// This should be an array with keys set to feature classnames and arguments.
-	$features = [
-		// Add initial features here.
-	];
-	$features = apply_filters( 'create_wordpress_plugin_features', $features );
-	Feature_Manager::add_features( $features );
+	// Add features here.
+	$plugin = new Group();
```

Comment:
Something that is lost by this approach is the ability to get access to the list of loaded features. Previously, you'd be able to do that using static methods on the feature manager class, but here, it's all wrapped up in the main function.

Some thoughts:

1. Is it important to be able to get access to a list of loaded features at all? Is it useful in tests or elsewhere in code?
2. This implementation also drops the filter - if we brought the filter back, would that be sufficient for "giving access" to the list of features, and implementers could store a reference to features at that point if they needed to?

An alternate proposal:

```suggestion
	$plugin = new Group(
		...apply_filters(
			'create_wordpress_plugin_features',
			[
				// Add initial features here by creating new instances of classes that implement the Feature interface.
			]
		)
	);
```

Reply:
> Something that is lost by this approach is the ability to get access to the list of loaded features. Previously, you'd be able to do that using static methods on the feature manager class, but here, it's all wrapped up in the main function.

That's true. But I'm somewhat skeptical of the value of that list.

> Some thoughts:
> 
> 1. Is it important to be able to get access to a list of loaded features at all? Is it useful in tests or elsewhere in code?

My question exactly 😄 I'm not sure that I see the use case yet.

> 2. This implementation also drops the filter - if we brought the filter back, would that be sufficient for "giving access" to the list of features, and implementers could store a reference to features at that point if they needed to?
> 
> An alternate proposal:

This version of the filter leaves me with the same questions that the current filter does. What's the value in allowing extenders to remove classes from the manifest of features? Doesn't that make it difficult for a plugin developer to predict the behavior of the plugin?

Similarly, what's the value in allowing extenders to add their own classes to the manifest? An extender could slip in any feature instance and have the plugin treat the feature like its own — with error handling, for example. It seems unusually permissive.

Reply:
[Alleyvate](https://github.com/alleyinteractive/wp-alleyvate) immediately comes to mind as a place where we _do_ want this list of features to be filtered. In that case, we're doing it by a different mechanism, but could convert it to be a simpler filter of all features vs. running filters on each individual feature being activated.

However, I'm not convinced of the need to make the filter part of what the plugin produces by default - implementers could always add in this functionality if they need it.

Reply:
> However, I'm not convinced of the need to make the filter part of what the plugin produces by default - implementers could always add in this functionality if they need it.

I'd agree.